tests/integration/cocoon/test_reference_portfolios.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+from pathlib import Path
+import pandas as pd
+import lusid
+from lusidtools import cocoon as cocoon
+from lusidtools import logger
+from lusidtools.cocoon.utilities import create_scope_id
+from parameterized import parameterized
+
+
+class CocoonTestsReferencePortfolios(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+        cls.portfolios_api = cls.api_factory.build(lusid.api.PortfoliosApi)
+        cls.unique_id = create_scope_id()
+        cls.logger = logger.LusidLogger("debug")
+        cls.scope = "ModelFundTest"
+
+    @parameterized.expand(
+        [
+            [
+                "Load a reference portfolio with required attributes only",
+                "data/reference-test.csv",
+                {"code": "FundCode", "display_name": "display_name"},
+                {
+                    "description": "description",
+                    "created": "created",
+                    "base_currency": "base_currency",
+                },
+                [],
+                [],
+            ],
+            [
+                "Load a reference portfolio with required attributes AND properties",
+                "data/reference-test.csv",
+                {"code": "FundCode", "display_name": "display_name"},
+                {
+                    "description": "description",
+                    "created": "created",
+                    "base_currency": "base_currency",
+                },
+                ["strategy", "custodian"],
+                [],
+            ],
```

Comment:
need a test case for failure case e.g. not supplying all mandatory params and maybe also adding duplicates in the source DataFrame

Reply:
I added:

(1) Test for mandatory params
(2) Test for duplicates - duplicates are handled by a function in the `cocoon.py` so passing duplicates should not return an error.